src/constants.ts

```diff
@@ -0,0 +1,16 @@
+import consola from 'consola';
+
+consola.log('process.env.DOMAIN', process.env.DOMAIN);
+consola.log('process.env.host', process.env.host);
+consola.log('process.env.HOST', process.env.HOST);
+consola.log(
+  'process.env.HEROKU_APP_DEFAULT_DOMAIN_NAME',
+  process.env.HEROKU_APP_DEFAULT_DOMAIN_NAME
+);
+
+export const DOMAIN =
+  process.env.HEROKU_APP_DEFAULT_DOMAIN_NAME ||
+  process.env.DOMAIN ||
+  (`localhost:${process.env.PORT}` as const);
```

Comment:
```suggestion
  (`localhost:${process.env.PORT || 3000}` as const);
```

Or better, have `PORT` be defined here, with that default value and then imported into app.ts